tests/testthat/test-python-threads.R

```diff
@@ -0,0 +1,40 @@
+
+
+test_that("py_allow_threads() can enable/disable background threads", {
+
+  file <- tempfile()
+  on.exit(unlink(file), add = TRUE)
+
+  write_to_file_from_thread <- py_run_string("
+def write_to_file_from_thread(path, lines):
+    from time import sleep, localtime, strftime
+
+    def write_to_file(path, lines):
+        sleep(.1) # don't try to run until we've had a chance to return to the R main thread
```

Comment:
Just to confirm, this is just a way to make sure that this thread pauses execution until we later try to invoke something in Python? (In general, users shouldn't need to use `sleep()` to ensure proper sequencing here?)

Reply:
This sleep is there to make sure that we've give a chance for the main thread to exit back into R from the python call. Otherwise, it's possible that the background thread can steal the GIL while the main thread is still evaluating Python code, which would lead to a failing test.